config/configcompression/compressiontype.go

```diff
@@ -24,18 +36,45 @@ func (ct *Type) IsCompressed() bool {
 	return *ct != typeEmpty && *ct != typeNone
 }
 
-func (ct *Type) UnmarshalText(in []byte) error {
-	typ := Type(in)
-	if typ == TypeGzip ||
-		typ == TypeZlib ||
-		typ == TypeDeflate ||
-		typ == TypeSnappy ||
-		typ == TypeZstd ||
-		typ == typeNone ||
-		typ == typeEmpty {
-		*ct = typ
+func (ct *TypeWithLevel) UnmarshalText(in []byte) error {
+	var compressionTyp Type
+	var level int
+	var err error
+	parts := strings.Split(string(in), "/")
+	compressionTyp = Type(parts[0])
+	level = zlib.DefaultCompression
+	if len(parts) > 1 {
```

Comment:
Let's fail if `len(parts) > 2`
```suggestion
	if len(parts) == 1 {
```

Reply:
I think you mean `len(parts) == 2`? because this condition is for when `/level` is specified.

Reply:
Yes, I meant `len(parts) == 2` for this condition, sorry